# ZOMATO DATA ANALYSIS

In [7]:
import pandas as pd

In [17]:
df = pd.read_csv("Zomato-data-.csv")

## DATA CLEANING

In [51]:
df.describe()

,rate,votes,approx_cost(for two people)
count,148.000000,148.000000,148.000000
mean,3.633108,264.810811,418.243243
std,0.402271,653.676951,223.085098
min,2.600000,0.000000,100.000000
25%,3.300000,6.750000,200.000000
50%,3.700000,43.500000,400.000000
75%,3.900000,221.750000,600.000000
max,4.600000,4884.000000,950.000000


In [50]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 148 entries, 0 to 147
Data columns (total 7 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   name                         148 non-null    object 
 1   online_order                 148 non-null    object 
 2   book_table                   148 non-null    object 
 3   rate                         148 non-null    float64
 4   votes                        148 non-null    int64  
 5   approx_cost(for two people)  148 non-null    int64  
 6   listed_in(type)              148 non-null    object 
dtypes: float64(1), int64(2), object(4)
memory usage: 8.2+ KB


In [32]:
df.head()

,name,online_order,book_table,rate,votes,approx_cost(for two people),listed_in(type)
0,Jalsa,Yes,Yes,4.1,775,800,Buffet
1,Spice Elephant,Yes,No,4.1,787,800,Buffet
2,San Churro Cafe,Yes,No,3.8,918,800,Buffet
3,Addhuri Udupi Bhojana,No,No,3.7,88,300,Buffet
4,Grand Village,No,No,3.8,166,600,Buffet


In [36]:
df.tail()

,name,online_order,book_table,rate,votes,approx_cost(for two people),listed_in(type)
143,Melting Melodies,No,No,3.3,0,100,Dining
144,New Indraprasta,No,No,3.3,0,150,Dining
145,Anna Kuteera,Yes,No,4.0,771,450,Dining
146,Darbar,No,No,3.0,98,800,Dining
147,Vijayalakshmi,Yes,No,3.9,47,200,Dining


In [33]:
df.shape

(148, 7)

In [35]:
df.ndim

2

In [48]:
df.columns

Index(['name', 'online_order', 'book_table', 'rate', 'votes',
       'approx_cost(for two people)', 'listed_in(type)'],
      dtype='object')

In [44]:
df.isnull().sum()

name                           0
online_order                   0
book_table                     0
rate                           0
votes                          0
approx_cost(for two people)    0
listed_in(type)                0
dtype: int64

In [27]:
df.duplicated().sum()

np.int64(0)

- None of the missing values are found
- None of the duplicated values are found
- Data set is clean and ready for Analysis

## DATA ANALYSIS

### 1. How many restaurants offer online ordering?

In [55]:
df["online_order"].value_counts()

online_order
No     90
Yes    58
Name: count, dtype: int64

- The number of restaurants offering online orders are 58

### 2. What is the average cost for 2 people?

In [64]:
df["approx_cost(for two people)"].mean()

np.float64(418.2432432432432)

- Average cost for 2 people is approximately 418 rupees

###  3. How many restaurants marked rating 4 and above 4?

In [71]:
count = 0 
for values in df["rate"]:
    if values >= 4:
            count = count+1

print (count) 

34


- The no of restaurants marked rating 4 and above 4 are 34

### 4 . What is the highest no of votes received by any restaurants?

In [75]:
highest  = 0
for values in df["votes"]:
    if values > highest:
         highest = values
print(highest)
    
    

4884


### 5. Find the average rating of restaurants in each listed type.

In [78]:
df.groupby("listed_in(type)")["rate"].mean()

listed_in(type)
Buffet    3.842857
Cafes     3.765217
Dining    3.571818
other     3.912500
Name: rate, dtype: float64

- The above are the average ratings for each listed type 

## Final summary
  Zomato dataset analysis has been performed , analysing key insights considering  ratings, no of votes, table bookings.
  key findings include:
  - Online orders are less recorded in comparison to dinings, which indicates that public are more interested more dinings
  - Average cost for restaurants is 418, which says that most of the restaurants are affordable
  - highly rated restaurants are very few, which indicates challenging customer satisfaction
  - As dining has recorded lowest average rating(3.57),there is more room for improvement in customer experience